Specifying data input in a dashboard or app URL (#3100)
SkylineCommunications/dataminer-docs: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/Feeds/Specifying_data_input_in_a_dashboard_URL.md

uid: Specifying_data_input_in_a_dashboard_URL
---

# Specifying data input in a dashboard or app URL

From DataMiner 10.2.11/10.3.0 onwards<!--RN 34261-->, you can specify default data for a low-code app via URL parameters. This way, you can for instance specify the elements, parameters, views, etc. that need to be selected in the app. You can pass the data using a [JSON object](#json-syntax) in the URL.

> If a dashboard has been configured with one or more feed components, it is possible to specify data input for these feeds in a dashboard URL as well. This way, you can immediately make the dashboard display specific data when it is opened. From DataMiner 10.2.0/10.2.2 onwards<!--RN 31833-->, you can pass the data using a JSON object in the URL. Prior to DataMiner 10.2.0/10.2.2, you can use the [legacy syntax](#legacy-syntax).

Reviewer: The legacy syntax can still be used and will always be supported

Author: ```suggestion
If a dashboard has been configured with one or more feed components, it is possible to specify data input for these feeds in a dashboard URL as well. This way, you can immediately make the dashboard display specific data when it is opened. From DataMiner 10.2.0/10.2.2 onwards<!--RN 31833-->, you can pass the data using a JSON object in the URL. Prior to DataMiner 10.2.0/10.2.2, you can use the [legacy syntax](#legacy-syntax), which continues to be supported in recent DataMiner versions for now as well.
```

```

> [!TIP]
> See [Examples of passing the data using a JSON object in the URL](#examples-passing-the-data-using-a-json-object-in-the-url)

- The version number is currently always 1.

  > [!IMPORTANT]

>   > In a low-code app URL, use "v" instead of "version" to specify the version number.

Reviewer: This is not correct. Version is also valid. Actually the version is currently ignored because we don't have any different versions yet. Omitting the "version" will still work. It becomes important when this json object would change in the future version and would need a migration.

Author: ```suggestion
  > In a low-code app URL, both "v" and "version" can be used to specify the version number.
```

I added this note because I noticed "v" was used in the low-code app example further down. Is this correction accurate?

Reviewer: the example is wrong, "version" does not get encoded to %22v%22. I checked the code and only "version" is used to write to the url, and currently not yet used to read from the url. So "version" is the only valid (currently ignored) syntax for both dashboards and LCA.

- When you provide data in the (optional) *feedAndSelect* item, that data will be interpreted as if it was passed using the [legacy syntax](#legacy-syntax) described below.

> - When you provide data in the (optional) *feed* item, that data will only be used in the URL feed. It will not be used to select items in selection boxes on the dashboard or app page.

Reviewer: The URL feed does not work in LCA yet. Although this is more an issue and we want this to work at one point.

Author: ```suggestion
- When you provide data in the (optional) *feed* item, that data will only be used in the URL feed. It will not be used to select items in selection boxes on the dashboard.

  > [!NOTE]
  > The (optional) *feed* item is currently only available for passing data input in a dashboard URL.
```

> - ] is encoded as %5D.
> - / is encoded as %2F.

#### Example 2: low-code app

The following example URL selects one default element on the initial page. The component ID is "1", and the element ID is "1/6"

```url
https://<dma>/<app-id>?data={"version":1,"components":[{"cid":1,"select":{"elements":["1/6"]}]}}
```

This is the encoded equivalent: `https://<dma>/<app-id>?data=%7B%22v%22:1,%22components%22:%5B%7B%22cid%22:1,%22select%22:%7B%22elements%22:%5B%221%2F6%22%5D%7D%5D%7D%7D`

> [!NOTE]
>
> - { is encoded as %7B
> - " is encoded as %22.
> - [ is encoded as %5B.
> - ] is encoded as %5D.
> - / is encoded as %2F.

> > - } is encoded as %7D

Reviewer: depending on what data is selected, you can also have a space " " which gets encoded as %20.
Eg. "select":{"object manager instances":["guid"]}
becomes: %22select%22:%7B%22object%20manager%20instances%22:%5B%22guid%22%5D%7D

Author: Instead of adding a list of the characters and their URL-encoded equivalents used in the examples, I've added a new section to add this information (including space = %20). See [commit 705d0de](https://github.com/SkylineCommunications/dataminer-docs/pull/3100/commits/705d0de66fea55733154d2596bd4d003dd89ca5f)

Reviewer: Even better!

`https://[DMA IP]/dashboard/#/MyDashboards/dashboard.dmadb?data=%7B%22version%22%3A1%2C%22components%22%3A%5B%7B%22cid%22%3A123%2C%22select%22%3A%7B%22elements%22%3A%5B%221%2F2%22%2C%221%2F8%22%2C%22212%2F123%22%5D%2C%22parameters%22%3A%5B%221%2F2%2F3%22%2C%221%2F4%2F6%22%5D%7D%7D%5D%7D`

#### Example 2: low-code app

The following example URL selects one default element on the initial page. The component ID is "1", and the element ID is "1/6"

```url
https://<dma>/<app-id>?data={"version":1,"components":[{"cid":1,"select":{"elements":["1/6"]}]}}
```

> This is the encoded equivalent: `https://<dma>/<app-id>?data=%7B%22version%22:1,%22components%22:%5B%7B%22cid%22:1,%22select%22:%7B%22elements%22:%5B%221%2F6%22%5D%7D%5D%7D%7D`

Author: I changed "%22v%22" to "%22version%22". Is this example correct now, @GillesBara?
